### ISSUE TITLE
Feature request: add Longest prefix match

### ISSUE THREAD

**Issue by DmitryMak** [rmandvikar/csharp-trie#1]
It would be really nice to have for certain scenarios: [Longest prefix match on Wikipedia](https://en.wikipedia.org/wiki/Longest_prefix_match)

Here is an example of implementation in Java, see
`public String getMatchingPrefix(String input)`
[method here](https://www.geeksforgeeks.org/longest-prefix-matching-a-trie-based-solution-in-java/#)

**Comment by rmandvikar** [comment]
I've added it in 591b297 `("trie: Add GetLongestPrefixMatch()", 2023-11-02)`, but I'm not sure of the name yet, and will mostly change it. Any suggestions? 

See [1.4.0-alpha0](https://github.com/rmandvikar/csharp-trie/releases/tag/v1.4.0-alpha0), and [1.4.0-alpha0](https://www.nuget.org/packages/rm.Trie/1.4.0-alpha0).

**Comment by DmitryMak** [comment]
Thank you so much! IMHO the method name is exactly what it should be because it is a well known algorithm. But the parameter name should not be 'prefix' because it is actually a full word and the trie itself is acting as a collection of prefixes (from which we choose the longest).

```
/// <summary>
/// Get longest matching prefix from the Trie.
/// </summary>
string GetLongestPrefixMatch(string word);
```

**Comment by DmitryMak** [comment]
Just a quick follow up, would you also be willing to add a corresponding TrieMap method? Same logic, but returning a value associated with the longest prefix, instead of the prefix itself. Something like this?

`ITrieMap<TValue>.GetValueByLongestPrefixMatch(String word)`

**Comment by rmandvikar** [comment]
I've added. Please verify. 

```cs
// ITrie
string GetLongestPrefixMatch(string word);

// ITrieMap<TValue>
KeyValuePair<string, TValue>? GetLongestPrefixMatch(string word);
```

See [1.4.0-alpha2](https://github.com/rmandvikar/csharp-trie/releases/tag/v1.4.0-alpha2), and [1.4.0-alpha2](https://www.nuget.org/packages/rm.Trie/1.4.0-alpha2).

**Comment by rmandvikar** [comment]
In case you're interested, I also added
```cs
// ITrieMap<TValue>
IEnumerable<KeyValuePair<string, TValue>> GetLongestKeyValuePairs();
IEnumerable<KeyValuePair<string, TValue>> GetShortestKeyValuePairs();
```

See [1.5.0-alpha0](https://github.com/rmandvikar/csharp-trie/releases/tag/v1.5.0-alpha0), and [1.5.0-alpha0](https://www.nuget.org/packages/rm.Trie/1.5.0-alpha0).

**Comment by DmitryMak** [comment]
Thank you so much for the quick turnaround! It looks like exactly what we need. As a minor side note, `String word` parameter is missing in the [readme file](https://github.com/rmandvikar/csharp-trie/edit/dev/README.md)

**Comment by rmandvikar** [comment]
Sure. I'll let the prerelease versions sit for a few days before cutting out release versions. Thanks.

**Comment by rmandvikar** [comment]
Released [1.4.0](https://www.nuget.org/packages/rm.Trie/1.4.0), and [1.5.0](https://www.nuget.org/packages/rm.Trie/1.5.0).
